test/util/cert_generation.go

```diff
@@ -99,14 +102,14 @@ func Sign(req, parent *x509.Certificate, signerPrivateKey interface{}) (*x509.Ce
 
 // Returns an SVID template with many default values set. Should be overwritten prior to
 // generating a new test SVID
-func defaultSVIDTemplate() *x509.Certificate {
+func defaultSVIDTemplate(clk clock.Clock) *x509.Certificate {
 	return &x509.Certificate{
 		Subject: pkix.Name{
 			Country:      []string{"US"},
 			Organization: []string{"SPIRE"},
 		},
-		NotBefore: time.Now(),
-		NotAfter:  time.Now().Add(1 * time.Hour),
+		NotBefore: clk.Now(),
+		NotAfter:  clk.Now().Add(1 * time.Hour),
```

Comment:
not your bug, but ideally, it seems crisper to calculate NotBefore and NotAfter on the same "now" timestamp.